rasterio/transform.py

```diff
@@ -226,12 +226,15 @@ def rowcol(transform, xs, ys, op=math.floor, precision=None):
         ys = [ys]
 
     if precision is None:
-        eps = sys.float_info.epsilon * (1.0 - 2.0 * op(0.1))
+        eps = sys.float_info.epsilon
     elif isinstance(precision, int):
-        eps = 10.0 ** -precision * (1.0 - 2.0 * op(0.1))
+        eps = 10.0 ** -precision
     else:
         eps = precision
 
+    if op(0.1) > 0.1:
```

Comment:
Reading the documentation, `op(0.1)` has to produce an integer, either 0 or 1 (anything else would break rasterio in weird ways). We have the opportunity to check if op is correct by checking that behavior.
```python
if op(0.1) == 1:
    eps = -eps
elif op(0.1) != 0:
    raise RuntimeError("op must map a float to an integer")
```
What do you think of this check? The alternative would be to limit `op` to one of `floor`, `ceil`, or `round`.

Reply:
But the current check does handle the `op=float` case. The the question is: can `op` return another float instead?

Reply:
Example:
```python
def op(x):
    return x + 0.1
```
I wonder if https://docs.python.org/3/library/math.html#math.nextafter could be used here.
```
invtransform * (x + math.nextafter(x, op(x)), y - math.nextafter(y, op(y))
```